packages/dds/tree/src/test/rebaserAxiomaticTests.ts

```diff
@@ -380,6 +654,196 @@ function verifyComposeAssociativity<TChangeset>(
 	assertDeepEqual(rightPartialCompositions.at(-1), singlyComposed);
 }
 
+function verifyComposeWithInverseEqualsEmpty<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const metadata = defaultRevisionMetadataFromChanges(edits);
+
+	for (const edit of edits) {
+		const change = tagChange(edit.change, edit.revision);
+		const changeset = fieldRebaser.compose(
+			change,
+			tagChange(fieldRebaser.invert(change), change.revision),
+			metadata,
+		);
+		const actualChange = makeAnonChange(changeset);
+		assert(fieldRebaser.isEmpty !== undefined);
+		assert(fieldRebaser.isEmpty(actualChange.change));
+	}
+}
+
+function verifyComposeWithEmptyIsNoOp<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const metadata = defaultRevisionMetadataFromChanges(edits);
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	for (const edit of edits) {
+		const emptyChange = tagChange(
+			fieldRebaser.createEmpty(),
+			"test" as unknown as SessionSpaceCompressedId,
+		);
+		const changeset = fieldRebaser.compose(edit, emptyChange, metadata);
+		const actualChange = makeAnonChange(changeset);
+		assertDeepEqual(actualChange, makeAnonChange(edit.change));
+	}
+}
+
+function verifyRebaseLeftDistributivity<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 2, "expected at least 2 edits");
+	const editsToRebaseOver = namedEditsToRebaseOver.map(({ changeset }) => changeset);
+	const revInfo = defaultRevInfosFromChanges(editsToRebaseOver);
+
+	const editB = editsToRebaseOver[0];
+	const editC = editsToRebaseOver[1];
+	const rebaseMetaData = rebaseRevisionMetadataFromInfo(revInfo, [
+		editB.revision,
+		editC.revision,
+	]);
+	const actualChange = makeAnonChange(
+		fieldRebaser.rebaseComposed(rebaseMetaData, edit.change, editB, editC),
+	);
+
+	const expectedChange = fieldRebaser.rebase(fieldRebaser.rebase(edit.change, editB), editC);
+
+	assertDeepEqual(actualChange, makeAnonChange(expectedChange));
+}
+
+function verifyRebaseRightDistributivity<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	namedEditToRebaseOver: NamedChangeset<TChangeset>,
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(edits.length === 2, "expected 2 edits");
```

Comment:
Same comment as for left distributivity.

Reply:
Left alone for now, as this is currently a skipped test.